File: Python/EX8.py
# Classe Empregado
class Empregado:
    def __init__(self, nome, cargo, salario):
        self.nome = nome
        self.cargo = cargo
        self.salario = salario

    def exibir_informacoes(self):
        print(f"Nome: {self.nome}, Cargo: {self.cargo}, Salário: R${self.salario:.2f}")


# Classe Empresa
class Empresa:
    def __init__(self, nome_empresa):
        self.nome_empresa = nome_empresa
        self.empregados = []  # Lista para armazenar os empregados da empresa

    def adicionar_empregado(self, empregado):
        self.empregados.append(empregado)

    def listar_empregados(self):
        print(f"Lista de empregados da empresa {self.nome_empresa}:")
        for empregado in self.empregados:
            empregado.exibir_informacoes()

    def calcular_folha_pagamento(self):
        total_salarios = sum(empregado.salario for empregado in self.empregados)
        print(f"Folha de pagamento total: R${total_salarios:.2f}")


# Exemplo de uso
if __name__ == "__main__":
    # Criando empregados
    emp1 = Empregado("Joaquim", "Desenvolvedor", 5000)
    emp2 = Empregado("Marisa", "Gerente", 7500)
    emp3 = Empregado("Pedro", "Designer", 4000)

    # Criando uma empresa e adicionando empregados
    empresa = Empresa("Tech Solutions")
    empresa.adicionar_empregado(emp1)
    empresa.adicionar_empregado(emp2)
    empresa.adicionar_empregado(emp3)

    # Exibindo a lista de empregados e calculando a folha de pagamento
    empresa.listar_empregados()
    empresa.calcular_folha_pagamento()

File: Java/EX8.java
import java.util.ArrayList;
import java.util.List;

// Classe Empregado
class Empregado {
    private String nome;
    private String cargo;
    private double salario;

    // Construtor
    public Empregado(String nome, String cargo, double salario) {
        this.nome = nome;
        this.cargo = cargo;
        this.salario = salario;
    }

    // Método para exibir informações do empregado
    public void exibirInformacoes() {
        System.out.printf("Nome: %s, Cargo: %s, Salário: R$%.2f%n", nome, cargo, salario);
    }

    // Getter para o salário
    public double getSalario() {
        return salario;
    }
}

// Classe Empresa
class Empresa {
    private String nomeEmpresa;
    private List<Empregado> empregados; // Lista para armazenar os empregados da empresa

    // Construtor
    public Empresa(String nomeEmpresa) {
        this.nomeEmpresa = nomeEmpresa;
        this.empregados = new ArrayList<>();
    }

    // Método para adicionar um empregado
    public void adicionarEmpregado(Empregado empregado) {
        empregados.add(empregado);
    }

    // Método para listar os empregados da empresa
    public void listarEmpregados() {
        System.out.println("Lista de empregados da empresa " + nomeEmpresa + ":");
        for (Empregado empregado : empregados) {
            empregado.exibirInformacoes();
        }
    }

    // Método para calcular a folha de pagamento
    public void calcularFolhaPagamento() {
        double totalSalarios = 0;
        for (Empregado empregado : empregados) {
            totalSalarios += empregado.getSalario();
        }
        System.out.printf("Folha de pagamento total: R$%.2f%n", totalSalarios);
    }
}

// Classe principal com o método main
class Main4 {
    public static void main(String[] args) {
        // Criando empregados
        Empregado emp1 = new Empregado("Joaquim", "Desenvolvedor", 5000);
        Empregado emp2 = new Empregado("Marisa", "Gerente", 7500);
        Empregado emp3 = new Empregado("Pedro", "Designer", 4000);

        // Criando uma empresa e adicionando empregados
        Empresa empresa = new Empresa("Tech Solutions");
        empresa.adicionarEmpregado(emp1);
        empresa.adicionarEmpregado(emp2);
        empresa.adicionarEmpregado(emp3);

        // Exibindo a lista de empregados e calculando a folha de pagamento
        empresa.listarEmpregados();
        empresa.calcularFolhaPagamento();
    }
}